resources/assets/js/volumes/stores/sorters.js

```diff
@@ -106,22 +106,38 @@ let similaritySorter = {
         mixins: [SortComponent],
         data() {
             return {
-                fileIds: [],
+                indices: [],
                 title: 'Sort images by Similarity',
                 text: 'Similarity',
                 id: 'similarity',
+                cache: [],
             };
         },
         methods: {
             getSequence() {
-                // let ids = ;
+                return new Vue.Promise.resolve(this.indices);
 
-                // return new Vue.Promise.resolve(ids);
+            },
+            getSimilarityIndices() {
+                this.loadingSequence = true;
+                return VolumesApi.querySimilarityIndices({id: this.volumeId})
+                    .then(this.parseResponse)
+                    .then((sequence) => {
+                        this.cache = sequence;
+                        return sequence;
+                    })
+                    .finally(() => this.loadingSequence = false);
+            },
+            parseResponse(response) {
+                return response.data;
+            },
+            compare(a, b) {
+                return a - b;
             },
 
         },
         created() {
-            this.fileIds = biigle.$require('volumes.fileIds');
+            this.indices = this.getSimilarityIndices();
```

Comment:
This should be called in `getSequence`. Otherwise the indices would be always fetched when the volume overview is opened and not only if the sorter is activated.

Reply:
Good to know,. 
Another Question is, getSequence is called one time for a volume and return the sorted list of Image ids, right?
But I don't need to care about the call?

Reply:
`getSequence` is called each time the sorter is activated (hence it's a good idea to implement caching of the sequence). You only have to provide the function, the actual call is handled by the existing UI code.